app/models/offer.rb

```diff
@@ -0,0 +1,8 @@
+class Offer < ApplicationRecord
+  belongs_to :buyer, class_name: "User", foreign_key: :buyer_id
+
+  validates :description, :status, :final_price, :final_delivery_time, presence: true
+  validates :status, inclusion: { in: ['Accepted', 'Rejected', 'Pending'] }
```

Comment:
This will suffice for now, but enums are usually used for such pieces of data:
https://blog.saeloun.com/2022/01/05/how-to-use-enums-in-rails